src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java

```diff
@@ -61,17 +72,72 @@
             message = "Internal Server Error"
         )
     })
-    @GetMapping(produces = APPLICATION_JSON_VALUE)
+    @GetMapping(
+        path = "/orgServices",
+        produces = APPLICATION_JSON_VALUE
+    )
     public ResponseEntity<Object> retrieveOrgServiceDetails(
         @RequestParam(value = "serviceCode", required = false) String serviceCode,
         @RequestParam(value = "ccdCaseType", required = false) String ccdCaseType,
         @RequestParam(value = "ccdServiceNames", required = false) String ccdServiceNames) {
         log.info("Inside retrieveOrgServiceDetails");
-
         ConstraintValidation.validateInputParameters(serviceCode, ccdCaseType, ccdServiceNames);
         List<LrdOrgInfoServiceResponse> lrdOrgInfoServiceResponse = null;
         lrdOrgInfoServiceResponse = lrdService.retrieveOrgServiceDetails(serviceCode, ccdCaseType, ccdServiceNames);
         return ResponseEntity.status(200).body(lrdOrgInfoServiceResponse);
     }
 
+    @ApiOperation(
+        value = "This API will retrieve a Building Location's details for the given epims ID",
+        authorizations = {
+            @Authorization(value = "ServiceAuthorization"),
+            @Authorization(value = "Authorization")
+        }
+    )
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
+            message = "Successfully retrieved a Building Location's details",
+            response = LrdOrgInfoServiceResponse.class,
+            responseContainer = "list"
+        ),
+        @ApiResponse(
+            code = 400,
+            message = "Bad Request"
+        ),
+        @ApiResponse(
+            code = 401,
+            message = "Forbidden Error: Access denied"
+        ),
+        @ApiResponse(
+            code = 404,
+            message = "No Building Location found with the given ID"
+        ),
+        @ApiResponse(
+            code = 500,
+            message = "Internal Server Error"
+        )
+    })
+    @GetMapping(
+        path = "/building-locations/epims/{epims_id}",
+        produces = APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<Object> retrieveBuildingLocationDetailsByEpimsId(
+        @PathVariable(value = "epims_id") String epimsId) {
+
+        log.info("Obtaining building locations for epimm id: " + epimsId);
+
+        if (isEmpty(epimsId)) {
+            throw new InvalidRequestException("No epimm id provided");
+        }
+
+        if (!ValidationUtils.isStringInExpectedFormat(epimsId.strip(), NumericRegex)) {
+            throw new InvalidRequestException("epimm id is expected to be a number");
```

Comment:
This needs to be verified/confirmed by @kavithagade12 or Pradipta - is Epim ID numeric only?